src/configuration.cpp

```diff
@@ -0,0 +1,53 @@
+#include <optional>
+#include <nlohmann/json.hpp>
+
+#include "configuration.hpp"
+
+namespace
+{
+    nlohmann::json g_config;
+    std::optional<std::string> resource;
+    std::optional<uint64_t> read_buffer_size_in_bytes;
+    std::optional<uint64_t> write_buffer_size_in_bytes;
+} //namespace
+
+void irods::s3::set_config(const nlohmann::json& _config)
+{
+    g_config = _config;
```

Comment:
Would it be worth making this thread safe?

Reply:
Perhaps in the future. For now, we're choosing to keep the config static on program start to keep things simple and deterministic.